infrastructure/smart-contracts/contracts/upgrades/pool-factory/PoolFactory2.sol

```diff
@@ -821,4 +851,48 @@ contract PoolFactory2 is Initializable, AccessControlEnumerableUpgradeable {
     function setFailedKyc(address user, bool failed) external onlyRole(DEFAULT_ADMIN_ROLE) {
         failedKyc[user] = failed;
     }
+
+    /**
+    * @notice Retrieves the total amount of esXAI staked by a specific user across all interacted staking pools.
+    * @dev If the user's total stake has already been calculated and cached, the cached value is returned.
+    *      Otherwise, it calculates the total staked amount by iterating over all staking pools the user has interacted with.
+    * @param user The address of the user whose total staked amount is to be retrieved.
+    * @return The total amount of esXAI staked by the user across all pools.
+    */
+    function getTotalesXaiStakedByUser(address user) public view returns (uint256) {
+        if (totalStakeCalculated[user]) {
+            return _totalEsXaiStakeByUser[user];
+        }
+
+        uint256 totalStakeAmount = 0;
+        address[] memory userPools = interactedPoolsOfUser[user];
+        for (uint256 i = 0; i < userPools.length; i++) {
+            totalStakeAmount += StakingPool(userPools[i]).getStakedAmounts(user);
+        }
+        return totalStakeAmount;
+    }
+
+    /**
+    * @notice Calculates the total stake amount for a list of users across all pools they have interacted with.
+    * @dev Iterates over each user provided in the `users` calldata array. For each user, it checks if the total stake has already been calculated.
+    *      If not, it sums the staked amounts from all staking pools the user has interacted with and stores the result.
+    *      Uses storage for interacting pools array to optimize gas usage. Only processes users whose total stake hasn't been calculated yet.
+    * @param users An array of user addresses for which the total stake needs to be calculated.
+    */
+    function calculateUserTotalStake(address[] calldata users) external {
+        for (uint256 i = 0; i < users.length; i++) {
+            address user = users[i];
```

Comment:
Since we're using the `calldata` keyword for the `users` param we should move this assignment below the if statement, and update the if statement condition to check `users[i]` instead. This will save us from any unecessary copy instructions when the if statement is triggered.

Reply:
Done, thank you for the feedback!